R/visualization-functions.R

```diff
@@ -654,69 +654,192 @@ boxplot_vars <- function(md, include_vars, x_var) {
-  results <- dplyr::filter(results, .data$hgnc_symbol %in% c("XIST", "UTY"))
-  results <- tibble::rownames_to_column(results, var = "geneId")
-  results <- dplyr::left_join(results, sex_specific)
-  results <- tidyr::pivot_longer(
-    results,
-    -dplyr::all_of(c("geneId", "chromosome_name", "hgnc_symbol")),
-    names_to = "sampleId",
-    values_to = "counts(log)"
-    ) %>%
-    dplyr::mutate(
-      `counts(log)` = log(.data$`counts(log)`),
-      `counts(log)` = ifelse(
-        .data$`counts(log)` == -Inf, 0, .data$`counts(log)`
+plot_sexcheck <- function(clean_metadata, filtered_counts, biomart_results, sex_var) {
+  
+  #Initialize outputs
+  warning <- NULL
+  results <- NULL
+  sex_outliers <- NA
+  discordance_plots <- NA
+  if( 'UTY' %in% biomart_results$hgnc_symbol &
+      row.names(biomart_results[ biomart_results$hgnc_symbol %in% c('UTY') , ]) %in% row.names(filtered_counts) &
+      'XIST' %in% biomart_results$hgnc_symbol &
+      row.names(biomart_results[ biomart_results$hgnc_symbol %in% c('XIST') , ]) %in% row.names(filtered_counts) ){
+    
+    sex_counts <- filtered_counts[ row.names(filtered_counts) %in% row.names(biomart_results[ biomart_results$chromosome_name %in% c('X','Y') , ]), ]
+    
+    sex_pc <- stats::prcomp( limma::voom(
+      sex_counts),
+      scale. = TRUE,
+      center = TRUE)
+    
+    filt_pcs <- as.numeric( table( (sex_pc$sdev^2/sum(sex_pc$sdev^2)) >= 0.01 )['TRUE'] )
+    
+    #Correlate the PCs above 1 to Sex
+    scan_vars <- as.data.frame( sex_pc$rotation[,1:filt_pcs] )
+    #scan_vars$sex <- eval( parse( text = paste0( 'clean_metadata[ row.names(scan_vars),]$', sex_var ) ) )
+    scan_vars$xist <- t( filtered_counts[ row.names(biomart_results[ biomart_results$hgnc_symbol %in% 'XIST', ]), ] )
+    scan_vars$uty <- t( filtered_counts[ row.names(biomart_results[ biomart_results$hgnc_symbol %in% 'UTY', ]), ] )
+    
+    #Store Comps for plots 
+    xist_comp <- 0
+    xist_estimate <- 0
+    uty_comp <- 0
+    uty_estimate <- 0
+    for( i in 1:filt_pcs ){
+      
+      test_xist <- stats::cor.test( scan_vars[,i], scan_vars$xist, method='kendall' ) 
+      test_uty <- stats::cor.test( scan_vars[,i], scan_vars$uty, method='kendall' ) 
+      
+      if( test_xist$p.value < 0.05 ){
+        if( abs(test_xist$statistic) > xist_estimate ){
+          xist_estimate <- abs(test_xist$statistic)
+          xist_comp <- i
+        }
+      }else{}
+      if( test_uty$p.value < 0.05 ){
+        if( abs(test_uty$statistic) > uty_estimate ){
+          uty_estimate <- abs(test_uty$statistic)
+          uty_comp <- i
+        }
+      }else{}
+      #message( paste0( 'PC: ', i, " PVal: ", signif(test_xist$p.value, digits=3), " Raw Coeff: ", signif(abs(test_xist$statistic), digits=3)) )
+      #message( paste0( 'PC: ', i, " PVal: ", signif(test_uty$p.value, digits=3), " Raw Coeff: ", signif(abs(test_uty$statistic), digits=3)) )
+      
+    }
+    
+    #Are XIST and UTY Expression Aligned to the same PC
```

Comment:
Can you talk me through what is happening in this for statement? I think we can simplify it.